app/src/main/java/me/saket/dank/ui/accountmanager/AccountManagerActivity.java

```diff
@@ -0,0 +1,325 @@
+  }
+
+  @Override
+  @OnClick(R.id.account_manager_done)
+  public void finish() {
+    timerDisposable.dispose();
+    super.finish();
+  }
+
+  @OnClick(R.id.account_manager_logout)
+  public void logout() {
+    confirmAction(ACTION_LOGOUT);
+  }
+
+  private Completable queueToDelete(AccountManager account){
+    this.selectedAccount = account;
+    return this.confirmAction(ACTION_DELETE);
+  }
+
+  private Completable confirmAction(int action) {
+    if (confirmTimer.isDisposed()) {
+      ACTION = action;
+      int confirmText = ACTION_LOGOUT == ACTION ? R.string.userprofile_confirm_logout : R.string.userprofile_confirm_delete;
+
+      runOnUiThread(() -> {
+        // Stuff that updates the UI
+        logoutButton.setText(confirmText);
+        logoutButton.setVisibility(View.VISIBLE);
+      });
+
+
+      confirmTimer = Observable.timer(5, TimeUnit.SECONDS)
+          .compose(applySchedulers())
+          .subscribe(o -> {
+            runOnUiThread(() -> {
+              // Stuff that updates the UI
+              if (userSessionRepository.get().isUserLoggedIn()) {
+                logoutButton.setText(R.string.login_logout);
+                logoutButton.setVisibility(View.VISIBLE);
+              } else {
+                logoutButton.setText("");
+                logoutButton.setVisibility(View.INVISIBLE);
+              }
+            });
+          });
+
+    } else {
+      // Confirm logout/delete was visible when this button was clicked. Perform the action.
+      confirmTimer.dispose();
+      timerDisposable.dispose();
+
+      int ongoingActionText = ACTION_LOGOUT == ACTION ? R.string.userprofile_logging_out : R.string.userprofile_deleting_account;
+      runOnUiThread(() -> {
+
+        // Stuff that updates the UI
+        logoutButton.setText(ongoingActionText);
+        logoutButton.setVisibility(View.VISIBLE);
+      });
+
+      Thread thread = new Thread()
+      {
+        @Override
+        public void run() {
+          try {
+            while(true) {
+              sleep(2000);
+              runOnUiThread(() -> {
+                if (userSessionRepository.get().isUserLoggedIn()) {
+                  logoutButton.setText(R.string.login_logout);
+                  logoutButton.setVisibility(View.VISIBLE);
+                } else {
+                  logoutButton.setText("");
+                  logoutButton.setVisibility(View.INVISIBLE);
+                }
+              });
+            }
+          } catch (InterruptedException e) {
+            e.printStackTrace();
+          }
+        }
+      };
+
+      if (ACTION == ACTION_DELETE) {
+        timerDisposable = accountManagerRepository.get().delete(this.selectedAccount)
+            .subscribeOn(io())
+            .observeOn(mainThread())
+            .subscribe(
+                () -> {
+                  //this.userSessionRepository.get().switchAccount(null, getApplicationContext());
+                  thread.start();
```

Comment:
Mixing threads and rx, that asks for too much trouble :)

If I'm not mistaken you used it to delay the execution by 2 seconds. Did you try using `.delay(2, TimeUnit.SECONDS)` instead?

Reply:
Yeah. Rx and me... we don't get along very well haha. When the PR is updated I'll let someone take care of this issue.